metadata-ingestion/tests/integration/ldap/test_ldap.py

```diff
@@ -97,3 +97,66 @@ def test_ldap_memberof_ingest(docker_compose_runner, pytestconfig, tmp_path, moc
             output_path=tmp_path / "ldap_memberof_mces.json",
             golden_path=test_resources_dir / "ldap_memberof_mces_golden.json",
         )
+
+
+@pytest.mark.integration
+def test_ldap_config_stateful_ingest(
+    docker_compose_runner, pytestconfig, tmp_path, mock_time
+):
+    test_resources_dir = pytestconfig.rootpath / "tests/integration/ldap"
+
+    with docker_compose_runner(
+        test_resources_dir / "docker-compose.yml", "ldap"
+    ) as docker_services:
+        # The openldap container loads the sample data after exposing the port publicly. As such,
+        # we must wait a little bit extra to ensure that the sample data is loaded.
+        wait_for_port(docker_services, "openldap", 389)
+        time.sleep(5)
+
+        pipeline = Pipeline.create(
+            {
+                "run_id": "ldap-test",
+                "source": {
+                    "type": "ldap",
+                    "config": {
+                        "ldap_server": "ldap://localhost",
+                        "ldap_user": "cn=admin,dc=example,dc=org",
+                        "ldap_password": "admin",
+                        "base_dn": "dc=example,dc=org",
+                        "filter": "(memberOf=cn=HR Department,dc=example,dc=org)",
+                        "attrs_list": ["+", "*"],
+                        "group_attrs_map": {
+                            "members": "member",
+                        },
+                        "stateful_ingestion": {
+                            "enabled": "False",
+                            "remove_stale_metadata": "False",
+                            "fail_safe_threshold": "100",
+                            "state_provider": {
+                                "type": "datahub",
+                                "config": {
+                                    "datahub_api": {
+                                        "server": "http://localhost:8080",
```

Comment:
There isn't a live datahub instance running when the integration tests run, so this test will fail always fail in CI.

Given that much of this code is tested through other mechanisms, it's probably ok to not have an integration test for stateful ingestion.

Reply:
Completed